martin/src/source.rs

```diff
@@ -37,16 +38,17 @@ impl TileSources {
     pub fn get_catalog(&self) -> TileCatalog {
         self.0
             .iter()
-            .map(|(id, src)| (id.to_string(), src.get_catalog_entry()))
+            .map(|v| (v.key().to_string(), v.get_catalog_entry()))
             .collect()
     }
 
-    pub fn get_source(&self, id: &str) -> actix_web::Result<&dyn Source> {
+    pub fn get_source(&self, id: &str) -> actix_web::Result<TileInfoSource> {
         Ok(self
             .0
             .get(id)
             .ok_or_else(|| ErrorNotFound(format!("Source {id} does not exist")))?
-            .as_ref())
+            .value()
+            .clone())
     }
```

Comment:
Lets provide the rationale why the `Box` change was done:

The change to the boxed variant comes from here and cascades pretty far sadly.
Because accessing a value returns `Ref<..>` => a temporary variable

![image](https://github.com/user-attachments/assets/459a8a7d-aa39-4697-b4c5-9482db92207b)